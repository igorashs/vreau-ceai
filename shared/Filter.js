import styled, { css } from 'styled-components';
import CheckSvg from '@/icons/check.svg';
import { Button } from '@/shared/Button';

const StyledFilter = styled.div`
  display: flex;
  align-items: center;

  > *:focus {
    outline: none;
  }
`;

export const Filter = ({ label, text, checked, onChange }) => {
  return (
    <StyledFilter disableOutline={checked}>
      <Button
        icon={checked}
        onClick={onChange}
        aria-label={label}
        btnStyle="accent-text"
        noPadding
      >
        {checked && <CheckSvg />}
        {text}
      </Button>
    </StyledFilter>
  );
};
