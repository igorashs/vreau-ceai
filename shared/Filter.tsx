import styled from 'styled-components';
import CheckSvg from '@/icons/check.svg';
import Button from '@/shared/Button';

const StyledFilter = styled.div`
  display: flex;
  align-items: center;

  > *:focus {
    outline: none;
  }
`;

interface FilterProps {
  label?: string;
  text: string;
  checked?: boolean;
  onChange: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

export const Filter = ({
  label = '',
  text,
  checked = false,
  onChange,
}: FilterProps) => {
  return (
    <StyledFilter>
      <Button
        icon={checked}
        onClick={onChange}
        aria-label={label}
        btnStyle="accent-text"
        noPadding
      >
        {checked && <CheckSvg />}
        {text}
      </Button>
    </StyledFilter>
  );
};
