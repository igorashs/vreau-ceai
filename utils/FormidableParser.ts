import * as validator from '@/utils/validator';
import FormidableNameSpace, { Files, File, IncomingForm } from 'formidable';
import { NextApiRequest } from 'next';
import { Formidable, ProductFields } from 'types';
import { productMessages } from './validator/schemas/product';
import { NextApiSessionRequest } from './withSession';

type ParsedData = {
  fields: Partial<ProductFields>;
  files: Files & { src?: File };
};

type HandleProductForm = (
  req: NextApiRequest | NextApiSessionRequest,
  options: Partial<FormidableNameSpace.Options>,
  allowedFileTypes: string[],
) => Promise<ParsedData>;

const handleProductForm: HandleProductForm = async (
  req,
  options,
  allowedFileTypes,
) => {
  const data: ParsedData = await new Promise((resolve, reject) => {
    const form: Formidable = new IncomingForm(options);

    form.onPart = (part) => {
      if (part.filename) {
        // get file extension
        const fileType = part.mime?.split('/').pop() ?? '';

        // validate allowed file types
        if (!allowedFileTypes.includes(fileType)) {
          form.emit(
            'error',
            validator.createValidationError({
              message: productMessages.src.invalid,
              key: 'src',
            }),
          );
        }
      }

      form.handlePart(part);
    };

    form.parse(req, (err, fields, files) => {
      if (err) {
        // custom error on maxFileSize
        if (err.message.includes('maxFileSize')) {
          reject(
            validator.createValidationError({
              message: productMessages.src.max,
              key: 'src',
            }),
          );
        } else {
          reject(err);
        }
      }

      resolve({ fields, files });
    });
  });

  return data;
};

export default {
  handleProductForm,
};
