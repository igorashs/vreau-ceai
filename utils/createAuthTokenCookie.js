import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createAuthTokenCookie = (payload) => {
  const authToken = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  return cookie.serialize('auth-token', authToken, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });
};
