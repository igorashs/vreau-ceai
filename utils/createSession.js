import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createSession = (payload) => {
  const token = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  const [header, jwtPayload, signature] = token.split('.');
  const session = header + '.' + jwtPayload;

  const sessionCookie = cookie.serialize('session', session, {
    secure: process.env.NODE_ENV !== 'development',
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  const authCookie = cookie.serialize('auth', signature, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  return [sessionCookie, authCookie];
};
