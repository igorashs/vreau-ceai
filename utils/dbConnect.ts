import mongoose from 'mongoose';

const dbConnect = async () => {
  if (mongoose.connection.readyState >= 1) {
    return;
  }

  await mongoose.connect(process.env.MONGODB_URI as string, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useCreateIndex: true,
    useFindAndModify: false,
  });
};

export default dbConnect;
