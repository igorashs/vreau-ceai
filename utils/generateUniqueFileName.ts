import { nanoid } from 'nanoid';

const generateUniqueFileName = (fileName: string) => {
  // grab file extension
  const matches = fileName?.match(/\..+$/);

  // create new file name
  const uniqueFileName = nanoid() + (matches && matches[0]);

  return uniqueFileName;
};

export default generateUniqueFileName;
