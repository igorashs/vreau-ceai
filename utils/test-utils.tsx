import { render, RenderOptions } from '@testing-library/react';
import GlobalStyle from 'GlobalStyle';

const Wrapper: React.FC = ({ children }) => (
  <>
    <GlobalStyle />
    {children}
  </>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
