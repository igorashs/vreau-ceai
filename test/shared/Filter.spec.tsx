import { render, fireEvent } from '@/utils/test-utils';
import Filter from '@/shared/Filter';

describe('Filter', () => {
  it('renders correctly', () => {
    const onSearchMock = jest.fn();

    const { container, getByText } = render(
      <Filter text="filterOne" onChange={onSearchMock} />,
    );

    const btn = container.querySelector('button');

    expect(btn).toBeInTheDocument();
    if (btn) expect(fireEvent.click(btn));
    expect(onSearchMock).toBeCalledTimes(1);
    expect(getByText('filterOne')).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('checked prop', () => {
    it('renders with CheckSvg', () => {
      const onSearchMock = jest.fn();

      const { container } = render(
        <Filter text="filterOne" onChange={onSearchMock} checked />,
      );

      const svgIcon = container.querySelector('svg');
      expect(svgIcon).toBeInTheDocument();
    });
  });
});
