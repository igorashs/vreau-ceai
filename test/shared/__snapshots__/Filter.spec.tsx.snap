// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 7px;
  padding: calc(var(--baseline) / 4) calc(var(--baseline) / 2);
  border-radius: 4px;
  border: 0;
  color: var(--text-light);
  background-color: var(--accent-dark);
  box-shadow: 1px 1px 2px #00000066;
  text-transform: lowercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  background-color: transparent;
  text-transform: initial;
  box-shadow: none;
  color: var(--accent-text-dark);
  padding: 0;
}

.c1:hover {
  background-color: var(--accent);
}

.c1:hover {
  background-color: transparent;
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c1:disabled {
  background-color: transparent !important;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > *:focus {
  outline: none;
}

<div
  class="c0"
>
  <button
    aria-label=""
    class="c1"
  >
    filterOne
  </button>
</div>
`;
